src/HttpClient.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Jobcloud\Kafka\SchemaRegistryClient;
+
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $baseUrl;
+
+    /**
+     * @var string|null
+     */
+    private $username;
+
+    /**
+     * @var string|null
+     */
+    private $password;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $requestFactory;
+
+    /**
+     * @var ErrorHandlerInterface
+     */
+    private $errorHandler;
+
+    /**
+     * @param ClientInterface $client
+     * @param RequestFactoryInterface $requestFactory
+     * @param ErrorHandlerInterface $errorHandler
+     * @param string $baseUrl
+     * @param string|null $username
+     * @param string|null $password
+     */
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
+        ErrorHandlerInterface $errorHandler,
+        string $baseUrl,
+        ?string $username = null,
+        ?string $password = null
+    ) {
+        $this->client = $client;
+        $this->baseUrl = $baseUrl;
+        $this->username = $username;
+        $this->password = $password;
+        $this->requestFactory = $requestFactory;
+        $this->errorHandler = $errorHandler;
+    }
+
+    /**
+     * @param string $method
+     * @param string $uri
+     * @param array $body
+     * @param array $queryParams
+     * @return RequestInterface
+     */
+    private function createRequest(
+        string $method,
+        string $uri,
+        array $body = [],
+        array $queryParams = []
+    ): RequestInterface {
+
+        $queryString = 0 !== count($queryParams) ? '?' . http_build_query($queryParams) : '';
+
+        $url = $this->baseUrl . '/' . $uri . $queryString;
+
+        $request = $this->requestFactory->createRequest($method, $url);
+
+        if ([] !== $body) {
+            $jsonData = json_encode($body, JSON_THROW_ON_ERROR);
+
+            /** @var string $dataLength */
+            $dataLength = strlen($jsonData);
```

Comment:
```suggestion
            $dataLength = (string) strlen($jsonData);
```
would be more correct than the type hint i think, because strlen truly does return int

Reply:
Mutation will lower.